src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -188,6 +188,31 @@ public void ContentType_SetThenGetEmptyValue_ExpectNull(Uri remoteServer)
             });
         }
 
+        [Fact]
+        public async Task HttpWebRequest_SetHostHeader_ContainsPortNumber()
+        {
+            await LoopbackServer.CreateServerAsync(async (server, uri) =>
+            {
+                HttpWebRequest request = WebRequest.CreateHttp(uri);
+                string host = uri.Host + ":" + uri.Port;
+                request.Host = host;
+                Task<WebResponse> getResponse = request.GetResponseAsync();
+
+                await server.AcceptConnectionAsync(async connection =>
+                {
+                    await connection.Reader.ReadLineAsync(); // Skip status line.
+                    Assert.Contains(host, await connection.Reader.ReadLineAsync());
```

Comment:
This assumes the Host header will be the first header.  That's not guaranteed, is it?

You can use the ReadRequestHeadersAndSendResponse method, which will return to you a list of all of the received headers.  You can then Assert.Contains($"Host: {host}", headers), or something like that.

Reply:
> That's not guaranteed, is it?

I step through the code, you are right, Host is not guaranteed to be the first header. I will fix the test.

Reply:
>This assumes the Host header will be the first header. That's not guaranteed, is it?

The RFC has a "SHOULD" in terms of a client sending the 'Host:" header as the first header.  It is not a MUST.

Reply:
Sorry I want to correct my statement, I step through the wrong code (`HttpWebRequest request = WebRequest.CreateHttp(uri);`)

Actually, when we send a HWR (`HttpWebRequest.SendRequest()`), we will create a new `HttpRequestMessage` object, and the first header to be set is `Host` (If we explicitly set one.)

https://github.com/dotnet/corefx/blob/master/src/System.Net.Requests/src/System/Net/HttpWebRequest.cs#L1166

But I think using `ReadRequestHeadersAndSendResponse` can make test more robust.